libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1061,10 +1061,9 @@ bool SMTEncoder::visit(MemberAccess const& _memberAccess)
 
 	auto const& exprType = _memberAccess.expression().annotation().type;
 	solAssert(exprType, "");
-	auto identifier = dynamic_cast<Identifier const*>(&_memberAccess.expression());
 	if (exprType->category() == Type::Category::Magic)
 	{
-		if (identifier)
+		if (auto identifier = dynamic_cast<Identifier const*>(&_memberAccess.expression()))
```

Comment:
I generally prefer extracting const and pointer from auto. But would be fine if you leave as is :)
```suggestion
		if (auto const* identifier = dynamic_cast<Identifier const*>(&_memberAccess.expression()))
```

Reply:
Yep, should be that way